tests/func/metrics/test_common.py

```diff
@@ -2,28 +2,57 @@
 
 import pytest
 
+from dvc.main import main
 from tests.func.metrics.utils import _write_json
 
 
-def metrics_diff(dvc, filename, revision):
-    dvc.metrics.diff(targets=[filename], a_rev=revision)
-
-
-def plots_diff(dvc, filename, revision):
-    dvc.plots.diff(targets=[filename], revs=[revision])
-
-
 @pytest.mark.parametrize(
-    "diff_fun, metric_value",
-    ((metrics_diff, {"m": 1}), (plots_diff, [{"m": 1}, {"m": 2}])),
+    "command, metric_value",
+    (("metrics", {"m": 1}), ("plots", [{"m": 1}, {"m": 2}])),
 )
 def test_diff_no_file_on_target_rev(
-    tmp_dir, scm, dvc, caplog, diff_fun, metric_value
+    tmp_dir, scm, dvc, caplog, command, metric_value
 ):
     with tmp_dir.branch("new_branch", new=True):
         _write_json(tmp_dir, metric_value, "metric.json")
 
         with caplog.at_level(logging.WARNING, "dvc"):
-            diff_fun(dvc, "metric.json", "master")
+            assert (
+                main([command, "diff", "master", "--targets", "metric.json"])
+                == 0
+            )
 
     assert "'metric.json' was not found at: 'master'." in caplog.text
+
+
+@pytest.mark.parametrize(
+    "command, malformed_metric",
+    (("metrics", '{"m": 1'), ("plots", '[{"m": 1}, {"m": 2}'),),
+)
+def test_show_malformed_metric(
+    tmp_dir, scm, dvc, caplog, command, malformed_metric
+):
+    tmp_dir.gen("metric.json", malformed_metric)
+
+    with caplog.at_level(logging.ERROR, "dvc"):
+        assert main([command, "show", "metric.json"]) == 1
+
+    assert (
+        f"Could not parse {command} files. "
+        "Use `-v` option to see more details."
+    ) in caplog.text
+
+
+@pytest.mark.parametrize(
+    "command, run_options",
+    (("metrics", "-m/-M"), ("plots", "--plots/--plots-no-cache"),),
+)
+def test_show_no_metrics_files(tmp_dir, dvc, caplog, command, run_options):
+    with caplog.at_level(logging.ERROR, "dvc"):
+        assert main([command, "show"]) == 1
+
+    assert (
+        f"No {command} files in this repository. "
+        f"Use `{run_options}` options for "
+        f"`dvc run` to mark stage outputs as {command}."
+    ) in caplog.text
```

Comment:
Since this is really an exception that you are testing, it would be better to test it without going through CLI, as we did before. Just a bit cleaner.